book/src/dev/rfcs/0008-release-planning.md

```diff
@@ -0,0 +1,191 @@
+However, even for
+crates such as `zcashconsensus` that build native code, the `cargo`-managed
+build process is far, far more reliable than build processes for C or C++
+projects.  We would not be asking users to run autotools or `./configure`, just
+a one-step `cargo install`.  We also know that it's possible to reliably build
+Zebra on each platform with minimal additional steps, because we do so in our
+CI.
+
+In contrast to these downsides, distributing `zebra` through Cargo has a number
+of upsides.  First, because we distribute our libraries using crates.io, we
+already have to manage tooling for publishing to crates.io, so there's no
+additional work required to publish `zebrad` this way.  Second, we get a
+cross-platform update mechanism with no additional work, since `cargo install`
+will upgrade to the latest published version.  Third, we don't rely on any
+third parties to mediate the relationship between us and our users, so users
+can get updates as soon as we publish them.  Fourth, unlike a system package
+manager, we can pin exact hashes of every transitive dependency (via the
+`Cargo.lock`, which `cargo install` can be configured to respect).  Fifth,
+we're positioned to pick up (or contribute to) ecosystem-wide integrity
+improvements like a transparency log for `crates.io` or work on reproducible
+builds for Rust.
+
+This proposal is summarized above in the [guide-level
+explanation](#guide-level-explanation).
+
+## Release Processes
+
+The next question is what kind of release processes and automation we should
+use.  Here are two important priorities for these processes:
+
+1.  Reducing the friction of doing any individual release, allowing us to move
+    closer to a continuous deployment model;
+2.  Reducing the risk of error in the release process.
+
+These are roughly in order of priority but they're clearly related, since the
+more friction we have in the release process, the greater the risk of error,
+and the greater the risk of error, the more friction we require to prevent it.
+
+Automation helps to reduce friction and to reduce the risk of error, but
+although it helps enormously, it has declining returns after some point (in the
+sense that automating the final 5% of the work is significantly more complex
+and error-prone than automating the first 5% of the work).  So the challenge is
+to find the right "entry point" for the automated part of the system that
+strikes the right balance.
+
+One possibility is the following.  CD automation is triggered by pushing a new
+tag to the git repository.  Tags are specified as `crate-semver`, e.g.
+`zebra-network-3.2.8`, `zebrad-3.9.1`, etc.  When a new tag is pushed, the CD
+automation parses the tag to determine the crate name.  If it is `zebrad`, it
+builds a new Docker image and publishes the image and the crate.  Otherwise, it
+just publishes the crate.
+
+To publish a new version of any component crate, the process is:
```

Comment:
Here's what we did for the first alpha release:
1. write tag message
2. (squash and) merge readme PR
3. tag using git
4. close GitHub milestone
5. maybe tell some people

We used the following git commands to tag:
```
git fetch origin
git checkout main
git rebase origin/main
git tag -a v1.0.0-alpha.0 -m "<message>"
git push origin --tags
```

We should update the tag name in the README just before we tag a new release.

Other README updates should happen as needed, in any PRs that:
- change dependencies
- change features
- resolve known issues

But it would be good to double-check the README before we tag.